approov-service/src/main/java/io/approov/service/okhttp/ApproovService.java

```diff
@@ -166,8 +195,12 @@ public static synchronized void setDevKey(String devKey) throws ApproovException
      * @param header is the header to place the Approov token on
      * @param prefix is any prefix String for the Approov token header
      */
-    public static synchronized void setApproovHeader(String header, String prefix) {
+    public static synchronized void setApproovHeader(String header, String prefix) throws ApproovException {
         Log.d(TAG, "setApproovHeader " + header + ", " + prefix);
+        // We only allow this to proceed if message signing is not enabled since changing the header after configuring the message signing is not allowed
```

Comment:
No need to throw here anymore. The active approov header name is passed to the message signing config every time it processes a request.